lib/amqp_client.js

```diff
@@ -59,7 +60,8 @@ function AMQPClient(policy) {
   this.policy = u.deepMerge(this._originalPolicy);
   this._connection = null;
   this._session = null;
-
+  this._transportProvider = new TransportProvider();
```

Comment:
I don't think you need to store this, _or_ new it up, since it was implemented as a singletion. @mbroadst correct me if I'm wrong.